contracts/pcv/maple/MaplePCVDeposit.sol

```diff
@@ -0,0 +1,118 @@
+pragma solidity =0.8.13;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {IPool} from "./IPool.sol";
+import {IMplRewards} from "./IMplRewards.sol";
+import {CoreRef} from "../../refs/CoreRef.sol";
+import {PCVDeposit} from "../PCVDeposit.sol";
+
+/// @notice PCV Deposit for Maple
+/// Allows depositing only by privileged role to prevent lockup period being extended by griefers
+contract MaplePCVDeposit is PCVDeposit {
+    using SafeERC20 for IERC20;
+
+    /// @notice reference to the Maple Pool where deposits and withdraws will originate
+    IPool public immutable pool;
+
+    /// @notice reference to the Maple Staking Rewards Contract
+    IMplRewards public immutable mplRewards;
+
+    /// @notice reference to the underlying token
+    IERC20 public immutable token;
+
+    /// @notice scaling factor
+    uint256 public constant scalingFactor = 1e12;
+
+    /// @notice fetch underlying asset by calling pool and getting liquidity asset
+    /// @param _core reference to the Core contract
+    /// @param _pool reference to the Maple Pool contract
+    constructor(
+        address _core,
+        address _pool,
+        address _mplRewards
+    ) CoreRef(_core) {
+        pool = IPool(_pool);
+        token = IERC20(IPool(_pool).liquidityAsset());
+        mplRewards = IMplRewards(_mplRewards);
+    }
+
+    /// @notice return the amount of funds this contract owns in Maple FDT's
+    function balance() public view override returns (uint256) {
+        return pool.balanceOf(address(this)) / scalingFactor;
+    }
+
+    /// @notice return the underlying token denomination for this deposit
+    function balanceReportedIn() external view returns (address) {
+        return address(token);
+    }
+
+    /// @notice deposit PCV into Maple.
+    /// all deposits are subject to a minimum 90 day lockup,
+    /// no op if 0 token balance
+    /// deposits are then immediately staked to accrue MPL rewards
+    /// only pcv controller can deposit, as this contract would be vulnerable
+    /// to donation / griefing attacks if anyone could call deposit and extend lockup time
+    function deposit() external onlyPCVController {
+        uint256 amount = IERC20(token).balanceOf(address(this));
+        if (amount == 0) {
+            /// no op to prevent wasted gas
+            return;
+        }
+
+        /// pool deposit
+        token.approve(address(pool), amount);
+        pool.deposit(amount);
+
+        /// token withdraw
+        uint256 scaledDepositAmount = amount * scalingFactor;
+        pool.increaseCustodyAllowance(address(mplRewards), scaledDepositAmount);
+        mplRewards.stake(scaledDepositAmount);
+
+        emit Deposit(msg.sender, amount);
+    }
+
+    /// @notice function to start the cooldown process to withdraw
+    function signalIntentToWithdraw() external onlyPCVController {
+        pool.intendToWithdraw();
```

Comment:
Add this comment:

1. lp lockup on deposit --> 80 days locked up and can't withdraw
2. cool down period, where you call intend to withdraw --> you must wait 10 days before you can withdraw after calling function
3. after you've cooled down and past the lockup period, you have 2 days to withdraw before you have to restart the cool down period.